### DISCUSSION TITLE
Can I categorize or organize my snippets?

### DISCUSSION THREAD

**Discussion by rosie-at-pieces** [pieces-app/support#13] [Q&A]
Note: this discussion was created due to several users reaching out to us asking about this functionality.

**Comment by rosie-at-pieces** [comment]
@mack-at-pieces any thoughts here?

**Reply by mark-at-pieces** [reply]
Currently this is not a way to categorize or organize snippets, however what you can you is via search filter down search result pased on language/title, xyz...

That being said, we are thinking about 2 features. (both of which I will add as separate comments below) so we can upvote individually :)

**Reply by eumel1968** [reply]
That would be nice!

**Comment by mark-at-pieces** [comment]
Thoughts on the features:

1. collections, which would be the ability to group snippets into "folder"

**Reply by jorokolev** [reply]
Any time frame on the release of "Workspaces"?

**Reply by mack-at-pieces** [reply]
Hey @jorokolev Thanks for your interest in the Workspaces feature! We don't have a release date quite yet, but we'll be sure to notify this thread as soon as we do!

**Reply by Adil-Doka** [reply]
yaa! This feature will be great

**Comment by mark-at-pieces** [comment]
Thoughts on the features:

2. workspaces, the ability to switch to a different profile to view a group of different materials

**Reply by eumel1968** [reply]
That would be nice!

**Comment by phntmPwr** [comment]
It would also be nice to have an option to sort based on the application you are using the snippets with. I'm currently building out a collection of javascript expressions for After Effects which I'd like to separate from js snippets that I might use only with Obsidian, for example.

**Reply by eumel1968** [reply]
I agree with you

**Comment by peaklabs-dev** [comment]
Please add the flowing features in one of the next updates:

1. High-level organization with **workspaces**. These are completely separated and isolated environments to store snippets. --> They can be used to separate snippets in Work, Private, School and so on.
2. Let us create **projects** in a workspace. --> A WordPress project with PHP, JS, CSS snippets, a NextJS project...
3. In the workspace/project let us create **folders** (in the list view) -> Folders for example Security PHP Snippets, Styling Snippets, Admin Snippets and so on.
4. Inside the folders, let us create **subfolders** and **subsubfolders** to organize our snippets even deeper

**--> 1-5: Workspace > Project > Folder > Subfolder > subsubfolders are highly needed.**

Example: Workspace: Business > Project: WordPress Site Client 1 > PHP Security Snippets and another folder CSS Styling and Checkout Validation... > and some more subfolders...

**Reply by eumel1968** [reply]
I agree with you

**Comment by peaklabs-dev** [comment]
Also check out the discussion for organizational features: https://github.com/pieces-app/support/discussions/28.

**Comment by peaklabs-dev** [comment]
And once there are workspaces, make tags workspace global https://github.com/pieces-app/support/discussions/12.

**Comment by Arthur-Provost** [comment]
I would also really appreciate to have both folders and tags as they are complementary. Folders are very good for project management for instance but a part of the snippets could also go in several folders at the same time and tag is a good solution to that problem.

**Comment by freepaddler** [comment]
As heavy massCode user (with 300+ snippets with variants), that just started pieces.app experience:

+ +1 vote for folders (tags are pretty ok, may be a little too much)
+ snippets search should be improved to incremental (i guess fuzzy will be nice there), today search works for entire words only
+ after adding snippet and changing its code there is no option to re-annotate/re-tag it automatically, so snippet can have incorrect meta after depersonalisation. this is really a pain sometimes and even breaks search

**Comment by kstingel** [comment]
Any further updates on this?

**Comment by Arthur-Provost** [comment]
+1 for workspace features!
